src/test/java/teammates/test/cases/search/FeedbackResponseCommentSearchTest.java

```diff
@@ -0,0 +1,123 @@
+
+        FeedbackResponseCommentSearchResultBundle bundle = feedbackResponseCommentsDb.search("non-existent",
+                ins1OfCourse1);
+        verifySearchResults(bundle);
+        ______TS(
+                "success: search for feedbackresponsecomments; query string matches"
+                        + " some feedbackresponsecomments; results restricted based on instructor's privilege");
+
+        bundle = feedbackResponseCommentsDb.search("self feedback Question 2", ins1OfCourse1);
+        verifySearchResults(bundle, fdbrI1S1, fdbrI1S1Q2);
+
+        ______TS(
+                "success: search for feedbackresponsecomments; query string should be case-insensitive;results restricted "
+                        + "based on instructor's privilege");
+
+        bundle = feedbackResponseCommentsDb
+                .search("Instructor 1 COMMENT to student 1 self feedback Question 2", ins1OfCourse1);
+        verifySearchResults(bundle, fdbrI1S1Q2, fdbrI1S1);
+
+        ______TS(
+                "success: search for feedbackresponsecomments by searchable by courseId; results restricted "
+                        + "based on instructor's privilege");
+
+        bundle = feedbackResponseCommentsDb.search("idOfTypicalCourse1", ins1OfCourse1);
+        verifySearchResults(bundle, fdbrI1S1, fdbrI1S1Q2);
+
+        ______TS("success: search for students; deleted student no longer searchable");
+
+        feedbackResponseCommentsDb.deleteFeedbackResponseCommentsForCourse(fdbrI1S1Q3.courseId);
+        bundle = feedbackResponseCommentsDb.search("feedback Question 3", ins1OfCourse1);
+        verifySearchResults(bundle);
+
+    }
+
+    /*
+     * Verifies that search results match with expected output. Parameters are
+     * modified to standardize {@link FeedbackResponseCommentAttributes} for
+     * comparison.
+     *
+     * @param actual the results from the search query.
+     *
+     * @param expected the expected results for the search query.
+     */
+    private static void verifySearchResults(FeedbackResponseCommentSearchResultBundle actual,
+            FeedbackResponseCommentAttributes... expected) {
+        assertEquals(expected.length, actual.numberOfResults);
+        standardizedFeedbackResponseCommentsForComparison(expected);
+        List<FeedbackResponseCommentAttributes> feedbackCommentListNormalised =
+                new ArrayList<FeedbackResponseCommentAttributes>();
+        for (List<FeedbackResponseCommentAttributes> feedbackResponseCommentAttributesList : actual.comments
+                .values()) {
+            standardizedFeedbackResponseCommentsForComparison(feedbackResponseCommentAttributesList
+                    .toArray(
+                            new FeedbackResponseCommentAttributes[feedbackResponseCommentAttributesList
+                                                                  .size()]));
+            feedbackCommentListNormalised.add(feedbackResponseCommentAttributesList.get(0));
```

Comment:
@xpdavid Here the actual results returned is a list of list of FeedbackResponseCommentAttributes, after standardizing the results I'm adding the list of list of FeedbackResponseCommentAttributes to the first index of a single list of  FeedbackResponseCommentAttributes, for as to make expected result and actual result in the same form.. what do you think I should do instead? Any suggestions?